### PR TITLE
Users/zjrunner/nuget35rtm107

### PR DESCRIPTION
**Pull request by zjrunner**
Cherry picks from master with:
 * Nuget 3.5 (likely rtm, still calling it rc2 for now)
 * fixed typings and added UTs
 * fixups once typings were correct

### CODE REVIEW

**Comment by zjrunner** [comment]
build and test run from root, expected results on scenario tests on v2 and v1 agents:
https://codesharing-su0.visualstudio.com/VstsPackagingTestAutomation/_build/index?buildId=5248&_a=summary
https://codesharing-su0.visualstudio.com/VstsPackagingTestAutomation/_build/index?buildId=5247&_a=summary

**Comment by bryanmacfarlane** [comment]
Also don't forget to cherry pick over to master ...

File: Tasks/NuGetInstaller/task.loc.json [hunk 1]
```diff
@@ -90,7 +90,7 @@
       "groupName": "advanced",
       "options": {
         "3.3.0": "3.3.0",
-        "3.5.0.1829": "3.5.0 – build 1829 (rc2)",
+        "3.5.0.1829": "3.5.0 - build 1938 (rc2)",
```
Comment (zjrunner) [inline Tasks/NuGetInstaller/task.loc.json]: yep.  notice we changed the variable name from 'version' to 'uxOption' to help avoid confusion here.

**Comment by zjrunner** [comment]
To the point about merging to master, all these commits were cherry picks from master so we're good there.

**Pull request merged**